src/qmltypes/qmlfilter.cpp

```diff
@@ -531,6 +531,23 @@ void QmlFilter::setAnimateIn(int value)
     value = qBound(0, value, duration());
     if (value != m_service.time_to_frames(m_service.get(kShotcutAnimInProperty))) {
         m_service.set(kShotcutAnimInProperty, m_service.frames_to_time(value, mlt_time_clock));
+        if (value == 0 && m_service.time_to_frames(m_service.get(kShotcutAnimOutProperty)) == 0) {
+            // Clear simple keyframes
+            for (int i = 0; i < m_metadata->keyframes()->parameterCount(); i++) {
+                QString name = m_metadata->keyframes()->parameter(i)->property();
+                Mlt::Animation anim = getAnimation(name);
+                if (anim.is_valid() && anim.key_count() > 0) {
+                    QString value;
+                    if (anim.key_count() > 1) {
+                        value = m_service.anim_get(qUtf8Printable(name), 1);
+                    } else {
+                        value = m_service.anim_get(qUtf8Printable(name), 0);
+                    }
```

Comment:
I do not understand this logic. It gets the value for the first or second frame depending on the number of keyframes. That seems rather arbitrary. I think it should get the value at the current time (playhead) position: make it always look like I see it now. Same for `setAnimateOut()` below.

Reply:
The goal is to set the value to the "middle" simple keyframe value. I think from the user's perspective, it should be "I no longer want to fade in, so make it always be the middle value". In this case, the middle value will be the keyframe at index 1 (index 0 is the "start" simple keyframe). the index 0 code is just a failsafe that I never triggered in testing. I updated the code to clarify the goal of those lines.

Your comment also made me realize that I was not using the correct index to choose the middle keyframe for the animOut case. So I fixed that logic.

Reply:
Sorry, I was mistaken and had read it as get property value at position instead of an anim_get. This is fine.